Description: Fix typos in program source
Author: Anuradha Weeraman <anuradha@debian.org>
Last-Update: 2023-01-28
Forwarded: yes

--- a/src/Monitor.c
+++ b/src/Monitor.c
@@ -1477,7 +1477,7 @@ void *WaitForProfilerCompletion(void *thread_args /* struct ProcDumpConfiguratio
             }
            else if(status=='H')
             {
-                Trace("WaitForProfilerCompletion: Recieved health check ping from profiler");
+                Trace("WaitForProfilerCompletion: Received health check ping from profiler");
             }
 
             free(dump);
@@ -1524,4 +1524,4 @@ void *ProcessMonitor(void *thread_args /* struct ProcDumpConfiguration* */)
 
     Trace("ProcessMonitor: Exiting ProcessMonitor Thread");
     pthread_exit(NULL);
-}
\ No newline at end of file
+}
diff --git a/src/ProcDumpConfiguration.c b/src/ProcDumpConfiguration.c
index 470ec5b..1cb0e47 100644
--- a/src/ProcDumpConfiguration.c
+++ b/src/ProcDumpConfiguration.c
@@ -752,7 +752,7 @@ int PrintUsage()
     printf("   -fc     File descriptor count threshold above which to create a dump of the process.\n");
     printf("   -sig    Signal number to intercept to create a dump of the process.\n");
     printf("   -e      [.NET] Create dump when the process encounters an exception.\n");
-    printf("   -f      [.NET] Filter (include) on the (comma seperated) exception name(s).\n");
+    printf("   -f      [.NET] Filter (include) on the (comma separated) exception name(s).\n");
     printf("   -pf     Polling frequency.\n");
     printf("   -o      Overwrite existing dump file.\n");
     printf("   -log    Writes extended ProcDump tracing to syslog.\n");
