Description: Fix typos in program source
Author: Anuradha Weeraman <anuradha@debian.org>
Last-Update: 2023-10-15
Forwarded: yes

--- a/src/Monitor.c
+++ b/src/Monitor.c
@@ -387,7 +387,7 @@ void MonitorProcesses(struct ProcDumpConfiguration *self)
             pthread_mutex_lock(&queue_mutex);
 
             // TALQ_FOREACH does not support changing the list while iterating.
-            // We use a seperate delete list instead.
+            // We use a separate delete list instead.
             int count = 0;
             TAILQ_FOREACH(item, &configQueueHead, element)
             {
@@ -1723,7 +1723,7 @@ void *WaitForProfilerCompletion(void *thread_args /* struct ProcDumpConfiguratio
             }
            else if(status=='H')
             {
-                Trace("WaitForProfilerCompletion: Recieved health check ping from profiler");
+                Trace("WaitForProfilerCompletion: Received health check ping from profiler");
             }
 
             free(dump);
